Articles/FileSystemProviders/extending.md

```diff
@@ -0,0 +1,213 @@
+---
+v8-equivalent: "https://our.umbraco.com/documentation/Extending/FileSystemProviders/"
+versionFrom: 9.0.0
+verified-against: beta-1
+---
+
+# Custom file systems (IFileSystem)
+
+## Media Filesystem
+By default, Umbraco uses an instance of `PhysicalFileSystem` to handle the storage location of the media archive (wwwroot/media).
+
+This can be configured during composition:
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.DependencyInjection;
+using Umbraco.Cms.Core.Hosting;
+using Umbraco.Cms.Core.IO;
+using Umbraco.Cms.Infrastructure.DependencyInjection;
+
+namespace UmbracoExamples.Composition
+{
+    public class SetMediaFileSystemComposer : IUserComposer
+    {
+        public void Compose(IUmbracoBuilder builder)
+        {
+            builder.SetMediaFileSystem((factory) =>
+            {
+                IHostingEnvironment hostingEnvironment = factory.GetRequiredService<IHostingEnvironment>();
+                var folderLocation = "~/CustomMediaFolder";
+                var rootPath = hostingEnvironment.MapPathWebRoot(folderLocation);
+                var rootUrl = hostingEnvironment.ToAbsolute(folderLocation);
+
+                return new PhysicalFileSystem(
+                    factory.GetRequiredService<IIOHelper>(),
+                    hostingEnvironment,
+                    factory.GetRequiredService<ILogger<PhysicalFileSystem>>(),
+                    rootPath,
+                    rootUrl);
+            });
+        }
+    }
```

Comment:
I might be wrong, but isn't this sample missing a closing bracket at the end? 🤔 

```suggestion
    }
}
```

Reply:
Oops indeed it is, my bad 😅